### DISCUSSION TITLE
Does flux2 ignore a directory's subdirectories, as soon as there is a Kubernetes-kustomization.yml in it?

### DISCUSSION THREAD

**Discussion by johanneskastl** [fluxcd/flux2#4274] [Q&A]
I have seen this in a few repos in the wild, but have not found it in the docs (but might just have missed it).

Given a directory that is specified as base in a Flux2-Kustomization (`kustomize.toolkit.fluxcd.io/v1beta1`).
And given this directory has subdirectories.

In my tests it looked like all YAML files in all subdirectories are being used.

Is this still true, if there is a kustomization.yaml (Kubernetes, not Flux, i.e. `kustomize.config.k8s.io/v1beta1`) in it that does not contain all of the YAML files? Is this taken first and applied, which means one can exclude certain files by just not mentioning them in this kustomization?

Sorry if this has been asked and answered before. kustomization makes a terrible search term...

**Comment by somtochiama** [comment]
Yes, Flux stops looking at sub-directories when there's a `kustomization.yaml` in the current directory.
I can't see where this is explicitly mentioned in the docs, it's kind of hinted at here https://fluxcd.io/flux/components/kustomize/kustomizations/#generating-a-kustomizationyaml-file. I will check further and open a PR if there's no other mention of this.
